lib/datadog/core/configuration/settings.rb

```diff
@@ -634,7 +634,40 @@ def initialize(*_)
           #
           # @default `nil`.
           # @return [String,nil]
-          option :service
+          option :service do |o|
+            o.type :string, nilable: true
+          end
+
+          # Declare additional service names to fold under the main service
+          # name used by remote configuration. Use when integrations are
+          # configured to use names that differ from DD_SERVICE and should be
+          # considered as a single remote configuration controllable service.
+          #
+          # @default `nil`.
+          # @return [Array<String>,nil]
+          option :extra_services do |o|
+            o.setter do |v|
```

Comment:
You could use `o.type :array, nilable: true` that way you can remove the `elsif !v.is_a?(Array)` branch from the `setter` block

Reply:
I was under the impression that the `setter` would either override `o.type` behaviour or be called after.

Also somehow I kind of preferred having the logic fully laid out in a single place. With `o.type :array, nilable: true` it becomes a bit unclear to me how corner cases would operate.

That makes me think, maybe it should not even be `nilable` as well as `o.default` to `[]`, the idea being to guarantee that when calling the getter you get an `Array`, even if empty.